sktime/transformations/series/detrend/_detrend.py

```diff
@@ -116,8 +123,20 @@ def _fit(self, X, y=None):
         -------
         self: a fitted instance of the estimator
         """
+        if len(X) <= 10:
+            self.out_of_sample = False
```

Comment:
estimator parameters should never be changed after `__init__`, so you need to replace this with a local variable condition.

Alternatively, we just drop this block and warn the user only.

Reply:
Done! I will see to it.